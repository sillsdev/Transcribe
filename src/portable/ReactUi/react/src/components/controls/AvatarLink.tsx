import * as React from 'react';
import Avatar from 'react-avatar';
import { Link } from 'react-router-dom';
import './AvatarLink.sass';

interface IProps {
    id: string;
    name: string;
    select?: (id: string) => any;
    size?: string;
    target: string;
    uri: string;
    avatarShape?: boolean
};

class AvatarLink extends React.Component<IProps, object> {
    public render() {
        const { avatarShape, id, name, select=(() => undefined), size="64", target, uri } = this.props;
        return (
            <div id={id} className="AvatarLink">
                <Link to={target} onClick={select.bind(this, id)}>
                    <Avatar className="OnHover"  name={name} src={uri} size={size} round={(avatarShape !== undefined && !avatarShape)? avatarShape: true}/>
                    <br /> <br />
                    <div className="caption">{name}</div>
                </Link>
            </div>
        )
    }
};

export default AvatarLink;
