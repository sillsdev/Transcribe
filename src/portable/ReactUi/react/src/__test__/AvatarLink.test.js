/* jshint esversion: 6 */
import React from 'react';
import expect from 'expect';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import AvatarLink from '../components/controls/AvatarLink';

configure({ adapter: new Adapter() });

// Snapshot for AvatarLink
describe('>>>Control: Avatarlink --- Snapshot',()=>{
    const minProps = {
        id: "my id",
        name: "my name",
        target: "/root",
        uri: "images/myImage.png",
    };

    it('+++capturing Snapshot of AvatarLink', () => {
        const renderedValue =  renderer.create(<Router><AvatarLink {...minProps}/></Router>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
//*******************************************************************************************************
describe('>>>Control: AvatarLink', () => {
    let wrapper;
    const minProps = {
        id: "my id",
        name: "my name",
        target: "/root",
        uri: "images/myImage.png",
    };

    beforeEach(() => {
        wrapper = shallow(<AvatarLink {...minProps} />);
    })

    it('+++ renders without exploding', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('+++ activates link when clicked', () => {
        const onButtonClick = sinon.spy();
        const linkWrapper = shallow(<AvatarLink {...minProps} select={onButtonClick}/>)
        linkWrapper.find('Link').simulate('click');
        expect(onButtonClick.calledOnce).toEqual(true);
    });

    it('+++ sets size of avatar', () => {
        expect(wrapper.find('Link').get(0).props.children[0].props.size).toEqual('64')
    });
});
