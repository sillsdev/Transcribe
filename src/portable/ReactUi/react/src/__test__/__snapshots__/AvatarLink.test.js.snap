// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Control: Avatarlink --- Snapshot +++capturing Snapshot of AvatarLink 1`] = `
<div
  className="AvatarLink"
  id="my id"
>
  <a
    href="/root"
    onClick={[Function]}
  >
    <div
      className="OnHover sb-avatar"
      onClick={undefined}
      style={
        Object {
          "borderRadius": "100%",
          "display": "inline-block",
          "fontFamily": "Helvetica, Arial, sans-serif",
          "height": "64px",
          "verticalAlign": "middle",
          "width": "64px",
        }
      }
    >
      <div
        className="OnHover sb-avatar__text"
        style={
          Object {
            "background": undefined,
            "borderRadius": "100%",
            "color": "#FFF",
            "height": "64px",
            "lineHeight": "initial",
            "textAlign": "center",
            "textTransform": "uppercase",
            "width": "64px",
          }
        }
      >
        <div
          style={
            Object {
              "display": "table",
              "height": "100%",
              "tableLayout": "fixed",
              "width": "100%",
            }
          }
        >
          <span
            style={
              Object {
                "display": "table-cell",
                "fontSize": "100%",
                "verticalAlign": "middle",
                "whiteSpace": "nowrap",
              }
            }
          >
            <span />
          </span>
        </div>
      </div>
    </div>
    <br />
     
    <br />
    <div
      className="caption"
    >
      my name
    </div>
  </a>
</div>
`;
